### PR TITLE
CLDR-15328 add @ORDERED for icu:extension, remove restrictions on using @ORDERED for icu: elements

### PR DESCRIPTION
**Pull request by pedberg-icu**
CLDR-15328

- [ ] This PR completes the ticket.

Add the @ORDERED metadata item for the icu:extension element to generate a distinguished path, and remove obsolete restrictions on using @ORDERED with icu:-namespace elements.

### CODE REVIEW

File: tools/cldr-code/src/main/java/org/unicode/cldr/api/CldrPaths.java [hunk 1]
```diff
@@ -205,8 +205,9 @@ static boolean isLeafPath(CldrPath path) {
      * sort index).
      */
     static boolean isOrdered(CldrDataType dtdType, String elementName) {
-        // Elements with namespaces unknown the the DTD are never ordered.
-        if (elementName.indexOf(':') != -1) {
+        // Elements with namespaces unknown to the DTD are never ordered
+        // (but it knows about icu: elements).
+        if (elementName.indexOf(':') != -1 && !elementName.startsWith("icu:")) {
```
[the remark this answers is not in the document]
Comment (srl295) [reply tools/cldr-code/src/main/java/org/unicode/cldr/api/CldrPaths.java:210]: There might be test cases somewhere. But, end users might have their own namespace.

[the remark this answers is not in the document]
Comment (srl295) [reply tools/cldr-code/src/main/java/org/unicode/cldr/api/CldrPaths.java:210]: In my view, the major problem here is that `CldrDataType` (and its second cousin `DtdType`) is an enum at all, when the type should be driven entirely by the DTD and not a closed set. This is what https://unicode-org.atlassian.net/browse/CLDR-13130 would need to fix.

**Review by srl295 — APPROVED** [review]
I'm going to mark it as LGTM with opportunity for improving comments and docs.

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/DtdData.java [hunk 2]
```diff
@@ -1202,9 +1202,8 @@ public boolean isDeprecated(String elementName, String attributeName, String att
     public boolean isOrdered(String elementName) {
         Element element = nameToElement.get(elementName);
         if (element == null) {
-            if (elementName.startsWith("icu:")) {
-                return false;
-            }
+            // No longer need to always return false for icu: elements per CLDR-8614,
+            // nameToElement should now know about them (see getInstance above).
```
[the remark this answers is not in the document]
Comment (pedberg-icu) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/DtdData.java:1206]: And I have made that comment update in [PR-1773](https://github.com/unicode-org/cldr/pull/1773)

**Pull request merged**